dspace-server-webapp/src/main/java/org/dspace/app/rest/PrimaryBitstreamController.java

```diff
@@ -0,0 +1,161 @@
+
+    /**
+     * This method updates the primaryBitstream on the given Bundle.
+     * <br><code>
+     * curl -i -X PUT "http://{dspace.server.url}/api/core/bundles/{bundle-uuid}/primaryBitstream"
+     * -H "Content-type:text/uri-list"
+     * -d "https://{dspace.server.url}/api/core/bitstreams/{bitstream-uuid}"
+     * </code>
+     *
+     * @param uuid      The UUID of the Bundle of which the primaryBitstream will be updated
+     * @param request   The HttpServletRequest
+     * @return          The Bundle of which the primaryBitstream was updated
+     */
+    @PreAuthorize("hasPermission(#uuid, 'BUNDLE', 'WRITE')")
+    @RequestMapping(method = RequestMethod.PUT, consumes = {"text/uri-list"})
+    public BundleResource updatePrimaryBitstream(@PathVariable UUID uuid,
+                                                 HttpServletRequest request) {
+        Context context = ContextUtil.obtainContext(request);
+        BundleRest bundleRest = repository.updatePrimaryBitstream(context, uuid,
+                                                                  getBitstreamFromRequest(context, request),
+                                                                  utils.obtainProjection());
+        return converter.toResource(bundleRest);
+    }
+
+    /**
+     * This method deletes the primaryBitstream on the given Bundle.
+     * <br><code>
+     * curl -i -X DELETE "http://{dspace.server.url}/api/core/bundles/{bundle-uuid}/primaryBitstream"
+     * </code>
+     *
+     * @param uuid      The UUID of the Bundle of which the primaryBitstream will be deleted
+     * @param request   The HttpServletRequest
+     * @return          The Bundle of which the primaryBitstream was deleted
+     */
+    @PreAuthorize("hasPermission(#uuid, 'BUNDLE', 'WRITE')")
+    @RequestMapping(method = RequestMethod.DELETE)
+    public ResponseEntity<RepresentationModel<?>> deletePrimaryBitstream(@PathVariable UUID uuid,
+                                                                         HttpServletRequest request) {
+        Context context = ContextUtil.obtainContext(request);
+        repository.deletePrimaryBitstream(context, uuid);
+        return ControllerUtils.toEmptyResponse(HttpStatus.NO_CONTENT);
+    }
+
+    /**
+     * This method parses a URI from the request body and resolves it to a Bitstream.
+     *
+     * @param context   The current DSpace context
+     * @param request   The HttpServletRequest
+     * @return          The resolved Bitstream
+     */
+    private Bitstream getBitstreamFromRequest(Context context, HttpServletRequest request) {
+        List<DSpaceObject> dsoList = utils.constructDSpaceObjectList(context, utils.getStringListFromRequest(request));
+        if (dsoList.size() != 1 || dsoList.get(0).getType() != BITSTREAM) {
+            throw new UnprocessableEntityException("No bitstream has been specified " +
+                                                       "or the data cannot be resolved to a bitstream.");
```

Comment:
if you have time it would be nice to improve the message here, something like
A single bitstream must be provided. Data are not valid there are no or many bitstreams or the URI doesn't match an existing bitstream